e2e/e2e-scenario.sh

```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+source e2e-helper.sh
+
+debug() {
+    local retval
+    retval=0
+    mkdir -p $SCENARIO_NAME-logs
+    INSTANCE_ID="$(az vmss list-instances --name $VMSS_NAME -g $MC_RESOURCE_GROUP_NAME | jq -r '.[0].instanceId')"
+    PRIVATE_IP="$(az vmss nic list-vm-nics --vmss-name $VMSS_NAME -g $MC_RESOURCE_GROUP_NAME --instance-id $INSTANCE_ID | jq -r .[0].ipConfigurations[0].privateIpAddress)"
+    set +x
```

Comment:
let's try to avoid set -x in general.

if you want to log a lot by default, let's define a logexec function which is just 
```
function logexec() { 
    echo "$@"
    $@
}
``` 

perhaps? 

fine if we want to keep it here for now though.

Reply:
can take it up with refactor work.